src/lib/expression/binary_predicate_expression.cpp

```diff
@@ -0,0 +1,46 @@
+#include "binary_predicate_expression.hpp"
+
+#include <sstream>
+
+#include "boost/functional/hash.hpp"
+#include "constant_mappings.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+BinaryPredicateExpression::BinaryPredicateExpression(const PredicateCondition predicate_condition,
+                                                     const std::shared_ptr<AbstractExpression>& left_operand,
+                                                     const std::shared_ptr<AbstractExpression>& right_operand)
+    : AbstractPredicateExpression(predicate_condition, {left_operand, right_operand}) {}
+
+const std::shared_ptr<AbstractExpression>& BinaryPredicateExpression::left_operand() const { return arguments[0]; }
+
+const std::shared_ptr<AbstractExpression>& BinaryPredicateExpression::right_operand() const { return arguments[1]; }
+
+std::shared_ptr<BinaryPredicateExpression> BinaryPredicateExpression::flipped() const {
```

Comment:
Do you envision cases where you will still need the old expression? Otherwise, we don't have to create an object here and can simply call `flip()`.

Reply:
this was never called anyway, removed it